tools/loader/Loader.cpp

```diff
@@ -271,6 +280,73 @@ static Kinded::Kind getKindFromNodeName(llvm::StringRef nodeName) {
   GLOW_UNREACHABLE("Unknown node name.");
 }
 
+/// Helper to get the BackendKind type from a backend's \p name. Need to be
+/// updated if a new backend comes.
+static BackendKind getBackendKindFromName(std::string &name) {
+  const llvm::StringMap<BackendKind> mapping(
+      {{"CPU", BackendKind::CPU},
+       {"Interpreter", BackendKind::Interpreter},
+       {"OpenCL", BackendKind::OpenCL},
+       {"Habana", BackendKind::Habana}});
+  if (mapping.find(name) == mapping.end()) {
+    GLOW_UNREACHABLE("Unknown backendKind name.");
+  }
+  return mapping.lookup(name);
+}
+
+/// Helper to get the parameters in DeviceConfig from \p str. The \p str has
+/// multiple lines, and each line with this format : "str1" : "str2".
+static llvm::StringMap<std::string> getBackendParams(std::string &str) {
+  llvm::StringMap<std::string> ret{};
+  std::string s;
+  std::istringstream f(str.c_str());
+  while (getline(f, s, '\n')) {
+    // Abstract the mapping from each line's string:
+    // ""str1" : "str2"" => ret["str1"] = "str2";
+    size_t pos1, pos2, pos3, pos4;
+    pos1 = s.find('"');
+    assert(pos1 != std::string::npos && "invalid string format");
+    pos2 = s.find('"', pos1 + 1);
+    assert(pos2 != std::string::npos && "invalid string format");
+    pos3 = s.find('"', pos2 + 1);
+    assert(pos3 != std::string::npos && "invalid string format");
+    pos4 = s.find('"', pos3 + 1);
+    assert(pos4 != std::string::npos && "invalid string format");
+    ret[s.substr(pos1 + 1, pos2 - pos1 - 1)] =
+        s.substr(pos3 + 1, pos4 - pos3 - 1);
+  }
+  return ret;
+}
+
+/// If the device config file \p loadDeviceDoncfigsFile available, load \p
+/// configs from the file. Otherwise, create \p numDevices number of devices
+/// based on \p backendKind.
+static void generateDeviceConfigs(
+    std::vector<std::unique_ptr<runtime::DeviceConfig>> &configs,
```

Comment:
can generateDeviceConfigs return the vector of configs? That way we don't have to pass in a reference.

Reply:
Done! Thanks!